packages/nuxt/ssr-plugin.mjs

```diff
@@ -0,0 +1,48 @@
+import { setSSRContext } from '@vueuse/core'
+import { useMeta } from '#meta'
+import { useCookie } from '#app'
+
+export default () => {
+  const cookieMap = new Map()
+  setSSRContext('getDefaultStorage', () => {
+    return {
+      getItem: (key) => {
+        if (!cookieMap.get(key))
+          cookieMap.set(key, useCookie(key))
+        return cookieMap.get(key).value
+      },
+      setItem: (key, value) => {
+        if (!cookieMap.get(key))
+          cookieMap.set(key, useCookie(key))
+        cookieMap.get(key).value = value
+      },
+      removeItem: (key) => {
+        if (!cookieMap.get(key))
+          cookieMap.set(key, useCookie(key))
+        cookieMap.get(key).value = undefined
+      },
+    }
+  })
+
+  if (process.server) {
+    setSSRContext('updateHTMLAttrs', (selector, attr, value) => {
```

Comment:
This is pinning ssr context to a global shared variable. context should available in non-async executation and unset afterwards.